### ISSUE TITLE
unable to scroll on touch screen devices when the dndzone fills the page

### ISSUE THREAD

**Issue by NotLatif** [isaacHagoel/svelte-dnd-action#613]
Hi,

Touch input users are unable to scroll page when a dndzone fills it.

eg: [https://www.youtube.com/shorts/mDiEBc7fLw0](https://www.youtube.com/shorts/mDiEBc7fLw0)

Maybe an input delay would be a good workaround (you hold the item for x seconds before the drag action starts)

I tried doing it myself wrapping my dndzone in a `<TouchDelayWrapper />` component made like this
```svelte
<script>
    let touchTimeout;
    export let isDraggable = false;
    
    function handleTouchStart() {
      touchTimeout = setTimeout(() => {
        isDraggable = true;
        console.log('Touch hold');
      }, 500);
    }
  
    function handleTouchEnd() {
      clearTimeout(touchTimeout);
      isDraggable = false;
    }
  </script>
  
  <div 
    on:touchstart={handleTouchStart}
    on:touchend={handleTouchEnd}
    on:touchcancel={handleTouchEnd}
    style="touch-action: {isDraggable ? 'none' : 'pan-y pinch-zoom'}"
  >
    <slot {isDraggable}></slot>
  </div>
  ```
  
  but I could not seem to find a way to update my dndzone reactively
  ```svelte
  <script>
    //...bla bla bla
    let parentDraggable = false;
</script>

<TouchDelayWrapper bind:isDraggable={parentDraggable}>

    <!-- this updates reactively -->
    <h1>{parentDraggable}</h1> 

    <!-- this does not -->
    <div use:dndzone={{items, flipDuration, dragDisabled: !parentDraggable}}>
        
    </div>
</TouchDelayWrapper>
```

**Comment by isaacHagoel** [comment]
Would be great if you could make a minimal REPL for me to have a look at.
Thx

**Comment by NotLatif** [comment]
thanks for the quick response,

here: [https://svelte.dev/playground/57847eba179f4fd1ae736fc4554c0700?version=3.59.2](https://svelte.dev/playground/57847eba179f4fd1ae736fc4554c0700?version=3.59.2)

try to scroll through the list using touch input

**Comment by NotLatif** [comment]
Ok so I was able to find this example [REPL](https://svelte.dev/playground/7e28249932d84d59a9d01504bf14417e?version=3.44.3)
but it didn't handle touch inputs, so I edited it [REPL if someone needs it](https://svelte.dev/playground/f56bb9929209473db4b5522d1e6c907b?version=3.44.3)

It's not perfect but it's fine for now 

Issues with my REPL (skill issue)
-  text tries to gets selected on mobile for long presses,
- sometimes the page scrolls with the item you're dragging) 
 
I'll maybe come back another day to fix those

(maybe my use case is too specific) but otherwise it would be nice if svelte-dnd-action handled this someway for touch input users

**Comment by isaacHagoel** [comment]
Hi,
Thanks for the repl. Why not leave some room next to that the user can use
to scroll the page (common pattern)?

**Comment by NotLatif** [comment]
that's what I'm currently doing, it just looks better when it's full page wide for my current layout [(yt shorts example)](https://www.youtube.com/shorts/mDiEBc7fLw0)

**Comment by isaacHagoel** [comment]
Thanks @Tyler-Petrov 
I would start by looking here https://github.com/isaacHagoel/svelte-dnd-action/blob/master/src/pointerAction.js#L386
As you see there is already a function that avoids starting the drag until a certain condition is met (currently that the pointer has moved a few pixels).
So first try to hardcode something there. If it works passing a new option just involved updating the options in action.js and pointerAction.js

**Comment by Tyler-Petrov** [comment]
I'll take a look at it, thank you

**Comment by Tyler-Petrov** [comment]
I forked your repo, but I didn't notice any way to test it locally while I develop.  What's your work flow?  Should I add a small sveltekit app as well so testing is easier for others who are looking to contribute?  I was thinking of something along the lines of [sveltekit-superforms](https://github.com/ciscoheat/sveltekit-superforms/tree/main/src).

P.S.  I'm not trying to take over the development workflow in any way, but I'm just curious and offering options that make sense to me.  I'd love to hear your thoughts

**Comment by isaacHagoel** [comment]
Yes. I have my own set of examples I use to test locally but it's not
public. You can use a simple example and I will also run it through my
stuff before I merge

**Comment by Tyler-Petrov** [comment]
I got as far as adding the delay and the properties we talked about, but I couldn't get the scroll to register.  I realized that making a svelte wrapper for [Sortable.js](https://github.com/SortableJS/Sortable?tab=readme-ov-file#options) might be an easier than adding this functionality, so I might pursue that route.  I'm not sure where this Issue/PR goes from here, so let me know your thoughts

**Comment by isaacHagoel** [comment]
@Tyler-Petrov I implemented the feature. A bit more testing and I'll merge and release. if not relevant for you, other will benefit from it

**Comment by Tyler-Petrov** [comment]
@isaacHagoel I did find something else, but I love to see improvement either way.  Glad I could help (albeit a very small part) this feature!
